libcontainer/rootfs_linux.go

```diff
@@ -40,13 +40,12 @@ type mountConfig struct {
 // mountEntry contains mount data specific to a mount point.
 type mountEntry struct {
 	*configs.Mount
-	srcFD   string
-	idmapFD int
+	fd *int
```

Comment:
I was curious if (now that we no longer use magic `-1` values) we needed to use an unsigned-integer, but it  looks like `unix.MoveMount()` expects an `int`; https://github.com/golang/sys/blob/c406141231ada89c399c39dd52dd1e279c509f5c/unix/zsyscall_linux.go#L1272-L1288

```go
func MoveMount(fromDirfd int, fromPathName string, toDirfd int, toPathName string, flags int) (err error) {
```


That said, `mountViaFDs` does  construct the `/proc/self/fd/` path, so slightly wondering if we _should_ have something unsigned.

Any thoughts @kolyshkin @fuweid @rata ?

Reply:
@thaJeztah I'd use `*int`, fds are usually ints. I see we won't use the negative values now that we have a pointer to represent "don't use this", but if the functions we want to use take ints in any case, why would we want unsigned?

We might have bigger values stored in an unsiged int that we can represent with int (that is what several syscalls take). So, it seems like it can even create issues.

Am I missing something? I like `int*` :)

Reply:
No, not missing something directly, just thought that `/proc/self/fd/-123` would be invalid, and having something unsigned could save us having to deal with that 😂

Reply:
> /proc/self/fd/-123 would be invalid

Yeah. But it might be easy to debug I think 🤔